File: src/main/java/pl/uwr/server/model/Student.java
package pl.uwr.server.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;

@Data
@NoArgsConstructor
@Entity
public class Student {

    @Id
    private Long id;
    @OneToOne(mappedBy = "student", cascade = CascadeType.ALL)
    @PrimaryKeyJoinColumn
    private Token token;

    @OneToMany(mappedBy = "student")
    private List<Reservation> reservationList = new ArrayList<>();


    public Student(Long id, Token token){
        this.id = id;
        this.token = token;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Token getToken() {
        return token;
    }

    public void setToken(Token token) {
        this.token = token;
    }

    public List getReservationList() {
        return reservationList;
    }
}

File: src/main/java/pl/uwr/server/model/Student.kt
package pl.uwr.server.model

import lombok.NoArgsConstructor
import java.util.*
import javax.persistence.*

@Entity
@NoArgsConstructor
/**
 * klasa reprezentująca obiekty przechowywane w repozytorium rezerwacji
 */
class Student {
    @Id
            /**
             * ID studenta (jego login)
             */
    var id: Long
    @OneToOne(mappedBy = "student", cascade = [CascadeType.ALL])
    @PrimaryKeyJoinColumn
            /**
             * wygenerowany token studenta
             */
    var token: Token
    @OneToMany(mappedBy = "student")
            /**
             * lista rezerwacji studenta
             */
    val reservationList = mutableListOf<Reservation>()
    /**
     * pełne imię i nazwisko studenta. Wykorzystywane do rezerwacji cyklicznych
     */
    var fullName: String

    /**
     * konstruktor przyjmujący login użytkownika i wygenerowany token
     * @param id login użytkownika
     * @param token wygenerowany token przypisany do użytkownika
     */
    constructor(id: Long, token: Token) {
        this.id = id
        this.token = token
        fullName = ""
    }

    /**
     * konstruktor przyjmujący login użytkownika, wygenerowany token i imię użytkownika
     * @param id login użytkownika
     * @param token wygenerowany token przypisany do użytkownika
     * @param fullName imię i nazwisko użytkownika
     */
    constructor(id: Long, token: Token, fullName: String) {
        this.id = id
        this.token = token
        this.fullName = fullName
    }

    constructor(fullName: String) {
        this.id = -1L
        this.fullName = fullName
        token = Token(-1L, Calendar.getInstance(), "")
    }


}